src/cupbearer/detectors/extractors/activation_extractor.py

```diff
@@ -0,0 +1,34 @@
+from typing import Any, Callable
+
+import torch
+
+from cupbearer import utils
+from cupbearer.detectors.extractors.core import DictionaryExtractor
+
+
+class ActivationExtractor(DictionaryExtractor):
+    def __init__(
+        self,
+        names: list[str],
+        return_inputs: bool = False,
+        individual_processing_fn: Callable[[torch.Tensor, Any, str], torch.Tensor]
+        | None = None,
+        global_processing_fn: Callable[
+            [dict[str, torch.Tensor]], dict[str, torch.Tensor]
+        ]
+        | None = None,
+    ):
+        super().__init__(
+            feature_names=names,
+            individual_processing_fn=individual_processing_fn,
+            global_processing_fn=global_processing_fn,
+        )
+        self.names = names
+        self.return_inputs = return_inputs
+
+    def get_features(self, batch: Any) -> dict[str, torch.Tensor]:
+        inputs = utils.inputs_from_batch(batch)
+        acts = utils.get_activations(inputs, model=self.model, names=self.names)
+        if self.return_inputs:
```

Comment:
Possibly I should instead just hard-code that inputs are always passed along to `_compute_layerwise_scores`. Feels much simpler. The default extractor would then not be the identity but just `None` (and you don't get features by default)